Artifacts/windows-powershellcore/installPowerShellCore.ps1

```diff
@@ -0,0 +1,79 @@
+﻿param
+(
+    [Parameter(Mandatory = $true)]
+    [String]
+    $packageUrl,
+
+    [Parameter()]
+    [bool]
+    $installCRuntime = $false
+)
+
+if (-not (Split-Path -Path $packageUrl -Leaf).EndsWith('.msi'))
+{
+    throw "${packageUrl} is not the path to PowerShell Core MSI."
+}
+
+try
+{
+    $coreMSI = "${env:Temp}\PowerShellCore.msi"    
+    Write-Verbose -Message "Downloading ${packageUrl} to ${coreMSI} ..."
+    Invoke-WebRequest -Uri $packageUrl -OutFile $coreMSI -Verbose
+
+    if (Test-Path -Path $coreMSI)
+    {
+        if ($installCRuntime)
+        {
+            $osVersion = (Get-WmiObject -Class Win32_OperatingSystem).Name.Split('|')[0]
+            if ($osVersion -like "Microsoft Windows Server 2012 R2*")
+            {
+                Write-Verbose -Message 'Acquiring C runtime installer ...'
+                $ucArchive = "${env:Temp}\WindowsUCRT.zip"
+                $ucRuntimeUri = 'https://download.microsoft.com/download/3/1/1/311C06C1-F162-405C-B538-D9DC3A4007D1/WindowsUCRT.zip'
+                Invoke-WebRequest -Uri $ucRuntimeUri -OutFile $ucArchive
+
+                #Extract the archive
+                Add-Type -assembly "System.IO.Compression.FileSystem"
+                [IO.Compression.ZipFile]::ExtractToDirectory($ucArchive, "${env:Temp}\ucFiles")
+
+                #installing Universal C Runtime                
+                $msuPath = "${env:Temp}\ucFiles\Windows8.1-KB3118401-x64.msu"
+                Write-Verbose -Message "installing Universal C runtime from ${msuPath} ..."
+                $msuProcess = Start-Process -FilePath wusa.exe -ArgumentList "/install ${msuPath} /quiet" -Wait -PassThru
+                if (-not $msuProcess.ExitCode -eq 0)
```

Comment:
Change to
```powershell
if ($msuProcess.ExitCode -ne 0)
```

Reply:
You can also leverage parts of the following function to handle exit codes. Adjust to meet your needs.
```powershell
function Test-LastExitCode
{
    param
    (
        [string] $Message = ''
    )

    # Whenever we execvute commands such as '& somecommand' or via Invoke-Expression, we should always
    # check the exit code, so we can decide to stop processing at that time, by throwing an exception.
    $exitCode = $LASTEXITCODE
    if ($exitCode -and $exitCode -ne 0)
    {
        if ($Message)
        {
            if (-not $Message.EndsWith('.'))
            {
                $Message += '.'
            }
            $Message += ' '
        }
        $Message += "Last command exited with error code $exitCode"
        throw $Message
    }
}
```